src/internal/observable/dom/AjaxObservable.ts

```diff
@@ -71,28 +56,63 @@ export class AjaxObservable<T> extends Observable<T> {
-  // @ts-ignore: Property has no initializer and is not definitely assigned
   private xhr: XMLHttpRequest;
   private done: boolean = false;
+  public request: AjaxRequest;
 
-  constructor(destination: Subscriber<T>, public request: AjaxRequest) {
+  constructor(destination: Subscriber<T>, request: AjaxRequest) {
     super(destination);
 
-    const headers = (request.headers = request.headers || {});
+    // Normalize the headers. We're going to make them all lowercase, since
+    // Headers are case insenstive by design. This makes it easier to verify
+    // that we aren't setting or sending duplicates.
+    const headers: Record<string, any> = {};
+    const requestHeaders = request.headers;
+    if (requestHeaders) {
+      for (const key in requestHeaders) {
+        if (requestHeaders.hasOwnProperty(key)) {
+          headers[key.toLowerCase()] = requestHeaders[key];
+        }
+      }
+    }
 
-    // force CORS if requested
-    if (!request.crossDomain && !this.getHeader(headers, 'X-Requested-With')) {
-      (headers as any)['X-Requested-With'] = 'XMLHttpRequest';
+    // Set the x-requested-with header. This is a non-standard header that has
+    // come to be a defacto standard for HTTP requests sent by libraries and frameworks
+    // using XHR. However, we DO NOT want to set this if it is a CORS request. This is
+    // because sometimes this header can cause issues with CORS. To be clear,
+    // None of this is necessary, it's only being set because it's "the thing libraries do"
+    // Starting back as far as JQuery, and continuing with other libraries such as Angular 1,
+    // Axios, et al.
+    if (!request.crossDomain && !('x-requested-with' in headers)) {
+      headers['x-requested-with'] = 'XMLHttpRequest';
     }
 
-    // ensure content type is set
-    let contentTypeHeader = this.getHeader(headers, 'Content-Type');
-    if (!contentTypeHeader && typeof request.body !== 'undefined' && !isFormData(request.body)) {
-      (headers as any)['Content-Type'] = 'application/x-www-form-urlencoded; charset=UTF-8';
+    // Ensure content type is set
+    if (!('content-type' in headers) && request.body !== undefined && !isFormData(request.body)) {
+      headers['content-type'] = 'application/x-www-form-urlencoded; charset=UTF-8';
     }
 
-    // properly serialize body
-    request.body = this.serializeBody(request.body, this.getHeader(request.headers, 'Content-Type'));
+    const body: string | undefined = this.serializeBody(request.body, headers['content-type']);
+
+    this.request = {
+      // Default values
+      async: true,
+      crossDomain: true,
+      withCredentials: false,
+      method: 'GET',
+      responseType: 'json',
+      timeout: 0,
+
+      // Override with passed user values
+      ...request,
+
+      // Set values we ensured above
+      headers,
+      body,
+    };
+
+    // Create our XHR so we can get started.
+    this.xhr = request.createXHR ? request.createXHR() : new XMLHttpRequest();
```

Comment:
question: With this moved out of the `try`/`catch` - in `send` - what will happen with any errors effect here? What should happen?